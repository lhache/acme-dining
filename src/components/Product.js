import React, { Component } from 'react'
import './Product.css'

class Product extends Component {

  constructor(props) {
    super(props)
    this.state = {
      showDescription: false
    }

    this._showDescription = this._showDescription.bind(this)
  }

  _showDescription() {
    this.setState({showDescription: true})
  }

  render() {
    const { title, description } = this.props.product
    return (
      <div className="Product">
        <div className={this.state.showDescription ? 'b' : ''} onClick={this._showDescription}>
          {title}
        </div>
        {this.state.showDescription && <div>{description}</div>}
      </div>
    );
  }
}

export default Product
