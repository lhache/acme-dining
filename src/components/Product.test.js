import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Product from './Product';
import { withApollo } from 'react-apollo';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const product = {
  title: 'title',
  description: 'description'
}

it('renders without crashing', () => {
  const wrapper = shallow(<Product product={product} />)
  expect(wrapper).toMatchSnapshot();
});

it('renders with proper props', () => {
  const wrapper = shallow(<Product product={product} />)
  const inst = wrapper.instance();
  expect(inst.props.product).toBe(product)
});

it('should have the title with normal font style', () => {
  // meaning not bold
})

it('should have the description hidden', () => {

})

it('has default showDescription state equal to false', () => {
  const wrapper = shallow(<Product product={product} />)
  expect(wrapper.state().showDescription).toBe(false);
});

it('shows the description when the element is cliked', () => {

})

it('shows the title in bold when the element is cliked', () => {

})
