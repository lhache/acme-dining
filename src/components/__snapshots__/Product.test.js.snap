// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    product={
      Object {
        "description": "description",
        "title": "title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className=""
          onClick={[Function]}
        >
          title
        </div>,
        false,
      ],
      "className": "Product",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "title",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
            onClick={[Function]}
          >
            title
          </div>,
          false,
        ],
        "className": "Product",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "title",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
